### PR TITLE
Curate samples

### LINKED ISSUE #353: Curate samples
I believe it would be good to curate the samples before the rc1 is released. People will look at samples to get a grasp of shoes, therefore they should be as understandable as possible :-)

What do I mean by curate?
- **Make them clearer to understand** - rename some variables/methods or maybe even extract some methods or eventually add some comments.
- **make the sample names clearer** - nobody knows what `sample15` is, but with a better name one might know that. If you rename them make sure to also rename them in `samples/README`
- **Delete duplicated samples** - over time for one reason we amassed some samples that essentially do the same, find them and delete one of them! 

Thanks!

### PR DESCRIPTION
**Pull request by jasonrclark**
Fixes #353

At long last, I've plowed through the samples and done a bit of grooming.

Each change is a separate commit... could squash when we merge, but seemed easier to allow for backing anything out folks don't like.

At this point, all samples in the directory are good, and anything not is in `unsupported` so I stopped using the README as a list and started using it as a README :)

Made some minor color tweaks, dropped duplicates, and moved a few things into a `development` folder (for things like positioning as we go, which no one except me cares about the details of)

Also revised naming on the `good` samples so now we're in `simple`, `expert` and `nks` for all the files.

This gets the main sample load under 100, and looking a bit less random... ready for prime time!

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+```
+bundle exec rake samples:all
+```
+
+We have a couple directories:
+
+* `development` contains samples of interest to the Shoes team, but not much for everyone else
+* `packaging` contains subfolders so we can easily test packaging without picking up all the samples in the bundle.
+* `sounds` contains some samples with large audio files we also don't want bundled up in the `shoes` gem
+* `unsupported` contains a small set of samples that older versions of Shoes allowed but aren't anymore. These are kept for historical reference and with hope to someday move them out!
```
Comment (PragTob) [inline samples/README.md]: :+1:

File: samples/simple_displace.rb [hunk 2]
```diff
@@ -1,17 +1,100 @@
...
+            info "#{name} displaces to #{position}"
+            @objects[name].displace(*position)
+          end
+          button "replace" do
+            position = @layout.recall(name)
+            info "#{name} replaces to #{position}"
+            @objects[name].displace(*position)
+          end
+        end
+      end
```
Comment (PragTob) [inline samples/simple_displace.rb]: this... uhhh exploded in size?

Comment (jasonrclark) [reply samples/simple_displace.rb]: Odd. I suspect I removed the original simple_displace for some reason, then renamed over it later on. Will review the commits to make sure that was intentional.

Comment (jasonrclark) [reply samples/simple_displace.rb]: Confirmed... what's now `simple_displace.rb` was `good_displace.rb` before, and the old `simple_displace.rb` was a bouncing button of which we have another similar copy. Relevant commits:
* https://github.com/shoes/shoes4/pull/1556/commits/df8e791d50b7308ec54347c32b4ff86561fc5a51
* https://github.com/shoes/shoes4/pull/1556/commits/59866b3c970fcc7d79a80adfed2bb8af059123ed

Comment (PragTob) [reply samples/simple_displace.rb]: thanks!

**Comment by PragTob** [comment]
Good from my side, not sure if @jasonrclark still wants to do something here :)

**Comment by PragTob** [comment]
I'm gonna declare he doesn't and if not we can always make a new PR :sweat_smile:

**Comment by jasonrclark** [comment]
I was all set--just been a busy week!--so thank you for landing the PR @PragTob. ✨

**Pull request merged**